shaders/generate_shader_code.js

```diff
@@ -223,13 +220,20 @@ const args = (() => {
 
 
 // Generate shader source headers
+const shaderRoot = "shaders/";
 let generatedHeaders = [];
 let shaderNames = [];
-JSON.parse(fs.readFileSync(path.join(args.input, "manifest.json")))
+
+const targetRoot = path.join(args.root, "__generated__");
```

Comment:
Does these scripts still work for the CMake based in source files?

Reply:
It does, the invocation just changes. Before the script dumped everything into a flat directory and now it emits structured folders with everything where it should be. It will pack that in a `__generated__` folder  wherever you specify the root is. Still a manual copy from inside the `__generated__` folder required for CMake as long as we're checking generated code into the repository, but our goal should probably be to move away from that model.